packages/frontend/src/components/TableTree.tsx

```diff
@@ -118,43 +120,86 @@ const TableButtons: FC<{
+                icon="filter"
+                text="Add filter"
+                onClick={(e) => {
+                    onFilter();
+                    e.stopPropagation();
+                }}
+            />,
+        );
+    }
     return (
-        <div style={{ display: 'inline-flex', gap: '10px' }}>
-            <Popover2
-                isOpen={isOpen === undefined ? false : isOpen}
-                onInteraction={setIsOpen}
-                content={
-                    <Menu>
-                        <MenuItem
-                            icon={<Icon icon="console" />}
-                            text="Source"
-                            onClick={(e) => {
-                                if (source === undefined) {
-                                    return;
-                                }
-                                e.stopPropagation();
-                                onOpenSourceDialog(source);
-                                setIsOpen(false);
-                            }}
-                        />
-                    </Menu>
-                }
-                position={PopoverPosition.BOTTOM_LEFT}
-                lazy
-            >
-                <Tooltip2 content="View options">
-                    <Button
-                        minimal
-                        icon="more"
-                        onClick={(e) => {
-                            e.stopPropagation();
-                            setIsOpen(true);
-                        }}
-                    />
-                </Tooltip2>
-            </Popover2>
+        <div
+            style={{
+                display: 'inline-flex',
+                gap: '10px',
+                alignItems: 'center',
+                height: '30px',
+                width: '60px',
+            }}
+        >
+            {isFiltered ? (
+                <Icon icon="filter" />
+            ) : (
+                <div style={{ width: '16px' }} />
+            )}
+            {menuItems && isHovered ? (
+                <Popover2
+                    content={<Menu>{menuItems}</Menu>}
+                    autoFocus={false}
+                    position={PopoverPosition.BOTTOM_LEFT}
+                    minimal
+                    lazy
+                    interactionKind="click"
+                    renderTarget={({ isOpen, ref, ...targetProps }) => (
+                        <Tooltip2 content="View options">
+                            <Button
+                                {...targetProps}
+                                elementRef={ref === null ? undefined : ref}
+                                icon="more"
+                                minimal
+                                onClick={(e) => {
+                                    (targetProps as any).onClick(e);
+                                    e.stopPropagation();
+                                }}
+                            />
+                        </Tooltip2>
+                    )}
```

Comment:
This is getting so complicated. :/  I don't know if we are using blueprint components in the wrong way or if their library isn't good enough and forces us to implement so many hacks.

Reply:
The blueprint eject is coming